lib/nmatrix/shortcuts.rb

```diff
@@ -34,6 +34,92 @@
 
 class NMatrix
 
+  # Methods for generating magic matrix.
+  module MagicHelpers
+    class << self
+      def odd_magic(nm,shape)
+        row= shape-1
+        col= shape/2       
+        nm[row,col]= 1
+        (2..shape*shape).each do |i|
```

Comment:
What is `i` exactly? The variable name should exactly specify what's it going to do.